pyfive/high_level.py

```diff
@@ -217,8 +224,21 @@ def _get_object_by_address(self, obj_addr):
         """ Return the object pointed to by a given address. """
         if self._dataobjects.offset == obj_addr:
             return self
-        return self.visititems(
-            lambda x, y: y if y._dataobjects.offset == obj_addr else None)
+        
+        queue = deque([(self.name.rstrip('/'), self)])
+        while queue:
+            base, grp = queue.popleft()
+            for name, link_addr in grp._links.items():
+                full_path = f"{base}/{name}" if base else f"/{name}"
+                # check address without instantiating
+                if link_addr == obj_addr:
+                    # return instantiated object
+                    return grp[name]
+                # descend only if it's a subgroup (need to instantiate minimally)
+                # would be neat to have .is_group
+                dataobjs = DataObjects(self.file._fh, link_addr)
+                if not dataobjs.is_dataset and not dataobjs.is_datatype:
+                    queue.append((full_path, grp[name]))
```

Comment:
This gets the object by address without instantiating completely. The DataObject creation is needed, to check for group and step into groups. Not sure if this more performant, but it doesn't instantiate the complete Datasets.